skbio/tree/tests/test_tree.py

```diff
@@ -1303,6 +1303,42 @@ def test_shuffle_raises(self):
         with self.assertRaises(MissingNodeError):
             next(self.simple_t.shuffle(names=['x', 'y']))
 
+    def test_cache_leafcount(self):
+        t = TreeNode.read([r'((a,b)e,(c,d)f,g)r;'])
+        t._cache_leafcount()
+        self.assertEquals(t.find('r').leafcount, 5)
```

Comment:
These are generally useful properties. The issues aren't motivation to get them merged, its on ensuring the properties are correct and on smoothing out the API